client/webserver/webserver.go

```diff
@@ -240,26 +241,34 @@ func (s *WebServer) Run(ctx context.Context) {
 		return
 	}
 
-	// Close the listener on context cancellation.
+	// Shutdown the server on context cancellation.
+	var wg sync.WaitGroup
+	wg.Add(1)
 	go func() {
+		defer wg.Done()
 		<-ctx.Done()
-		err := listener.Close()
+		err := s.srv.Shutdown(context.Background())
 		if err != nil {
 			log.Errorf("Problem shutting down rpc: %v", err)
 		}
 	}()
+
 	log.Infof("Web server listening on %s", s.addr)
 	err = s.srv.Serve(listener)
 	if !errors.Is(err, http.ErrServerClosed) {
 		log.Warnf("unexpected (http.Server).Serve error: %v", err)
 	}
 	log.Infof("Web server off")
-	<-ctx.Done()
+
+	// Disconnect the websocket clients since Shutdown does not deal with
+	// hijacked websocket connections.
```

Comment:
Is "hijacked" the same as "upgraded"?

Reply:
If not precisely, that's at least the idea.  The http.Server.Shutdown docs use the following language:

> Shutdown does not attempt to close nor wait for hijacked connections such as WebSockets. The caller of Shutdown should separately notify such long-lived connections of shutdown and wait for them to close, if desired.